### PR TITLE
update types for node `node:test` 18.6.0

### PR DESCRIPTION
**Pull request by MoLow**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
see https://nodejs.org/api/test.html#describeit-syntax
and https://github.com/nodejs/node-core-test/pull/27

### CODE REVIEW

**Comment by yoursunny** [comment]
If the feature is only available since Node 18.6, should the version number in `index.d.ts` be updated?
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e9f2ce240289ea65b1a6485a48f6c951d0f5ddcd/types/node/index.d.ts#L1

**Comment by MoLow** [comment]
> If the feature is only available since Node 18.6, should the version number in `index.d.ts` be updated?

tried and got a linting error:
```
ERROR: 1:41   dt-header     Error parsing header. Expected: foo MAJOR.MINOR (patch version not allowed). See: https://github.com/microsoft/DefinitelyTyped-tools/blob/master/packages/dtslint/docs/dt-header.md
```

**Comment by yoursunny** [comment]
> ERROR: 1:41   dt-header     Error parsing header. Expected: foo MAJOR.MINOR (patch version not allowed).

It would take `18.6` but not `18.6.0`.

**Comment by MoLow** [comment]
not sure how to handle tests here - seems to be related to typescript version

**Review by mcollina — APPROVED** [review]
lgtm

File: types/node/test/util_types.ts [hunk 1]
```diff
@@ -60,7 +60,8 @@ if (types.isMap(object)) {
     object; // $ExpectType Map<unknown, unknown>
 
     if (types.isMap(readonlyMapOrRecord)) {
-        readonlyMapOrRecord; // $ExpectType ReadonlyMap<any, any>
+        // Test disabled since it fails in TypeScript 4.8 (intended) but not earlier.
```
Comment (peterblazejewicz) [inline types/node/test/util_types.ts:63]: /cc @sandersn same problem, microsoft/TypeScript#43825

**Review by peterblazejewicz — APPROVED** [review]
@MoLow all clear, it's just waiting for TS4.8 beta fallout to be reviewed

LGTM!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by MoLow** [comment]
Ready to merge

**Pull request merged**